### PR TITLE
Refactor imagediff and fix regression bug

### LINKED ISSUE #30683: New pull request imagediff js error
### Description

image

I commited single image and trying to make new pr. See the error

### Gitea Version

trunk

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

dokcer

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #30683
Regression of #30140

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
Where exactly was the bug?

**Comment by lunny** [comment]
> Where exactly was the bug?

https://github.com/go-gitea/gitea/issues/30683#issuecomment-2076469994

File: web_src/js/features/imagediff.js [hunk 1]
```diff
@@ -38,36 +38,36 @@ function getDefaultSvgBoundsIfUndefined(text, src) {
...
-    };
-  }
+function createContext(imageAfter, imageBefore) {
+  const sizeAfter = {
+    width: imageAfter && imageAfter.width || 0,
+    height: imageAfter && imageAfter.height || 0,
+  };
+  const sizeBefore = {
+    width: imageBefore && imageBefore.width || 0,
+    height: imageBefore && imageBefore.height || 0,
```
Comment (silverwind) [inline web_src/js/features/imagediff.js]:
```suggestion
    width: imageAfter?.width || 0,
    height: imageAfter?.height || 0,
  };
  const sizeBefore = {
    width: imageBefore?.width || 0,
    height: imageBefore?.height || 0,
```

**Comment by wxiaoguang** [comment]
> Where exactly was the bug?

In the issue's screenshot https://github.com/go-gitea/gitea/issues/30683#issue-2261038989

image

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
So the bug was basically a missed `?.`, right?

File: web_src/js/features/imagediff.js [hunk 2]
```diff
@@ -116,94 +116,96 @@ export function initImageDiff() {
...
+        if (boundsInfoAfterWidth) {
+          boundsInfoAfterWidth.textContent = `${sizes.imageAfter.naturalWidth}px`;
+          boundsInfoAfterWidth.classList.toggle('green', widthChanged);
+        }
         const boundsInfoAfterHeight = container.querySelector('.bounds-info-after .bounds-info-height');
-        boundsInfoAfterHeight.textContent = `${sizes.$image1[0].naturalHeight}px`;
-        if (heightChanged) boundsInfoAfterHeight.classList.add('green');
+        if (boundsInfoAfterHeight) {
+          boundsInfoAfterHeight.textContent = `${sizes.imageAfter.naturalHeight}px`;
+          boundsInfoAfterHeight?.classList.toggle('green', heightChanged);
```
Comment (wxiaoguang) [inline web_src/js/features/imagediff.js]:
```suggestion
          boundsInfoAfterHeight.classList.toggle('green', heightChanged);
```

**Comment by wxiaoguang** [comment]
> So the bug was basically a missed check if a queried element existed?

I think so.

**Pull request merged**